ci/build.sh

```diff
@@ -32,21 +33,43 @@ for i in ./fluent-plugin* ; do
   fi
 done
 
-echo "Building docker image with $DOCKER_TAG:$VERSION in `pwd`..."
+echo "Building docker image with $DOCKER_TAG:local in `pwd`..."
 cd ./deploy/docker
-docker build . -f ./Dockerfile -t $DOCKER_TAG:$VERSION --no-cache
+docker build . -f ./Dockerfile -t $DOCKER_TAG:local --no-cache
 rm -f ./gems/*.gem
 cd ../..
 
 echo "Test docker image locally..."
 ruby deploy/test/test_docker.rb
 
-if [ -z "$DOCKER_PASSWORD" ] || [ -z "$TRAVIS_TAG" ]; then
-    echo "Skip Docker pushing"
+if [ -n "$DOCKER_PASSWORD" ] && [ -n "$TRAVIS_TAG" ]; then
+  echo "Tagging docker image $DOCKER_TAG:local with $DOCKER_TAG:$VERSION..."
+  docker tag $DOCKER_TAG:local $DOCKER_TAG:$VERSION
+  echo "Pushing docker image $DOCKER_TAG:$VERSION..."
+  echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  docker push $DOCKER_TAG:$VERSION
+elif [ -n "$DOCKER_PASSWORD" ] && [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_EVENT_TYPE" == "push" ]; then
+  # Major.minor.patch version format
+  latest_release=`wget -q $DOCKER_TAGS -O - | jq -r .[].name | grep -v alpha | grep -v latest | sort --version-sort --field-separator=. | tail -1`
+  latest_major=`echo $latest_release | tr '.' $'\n' | sed -n 1p`
+  latest_minor=`echo $latest_release | tr '.' $'\n' | sed -n 2p`
+  latest_alpha=`wget -q $DOCKER_TAGS -O - | jq -r .[].name | grep alpha | grep "^$latest_major.$latest_minor" | sed 's/-alpha//g' | sort --version-sort --field-separator=. | tail -1`
+  if [ -n "$latest_alpha" ]; then
+    echo "Most recent release version: $latest_release, most recent alpha: $latest_alpha-alpha"
+  else
+    echo "Most recent release version: $latest_release, most recent alpha does not yet exist"
+    latest_alpha="$latest_release"
+  fi
+  new_patch=$((`echo $latest_alpha | tr '.' $'\n' | sed -n 3p`+1))
```

Comment:
In the case that we bumped the major or minor version, we could have a case like this: `latest_release=1.0.0`, `latest_alpha=0.0.10` in which case I believe `new_alpha` would be `1.0.11-alpha`, but we probably want to start the patch count from 0 again

Reply:
L56 also does `grep "^$latest_major.$latest_minor"` to find the most recent alpha for the most recent Major.minor release, so it will not find an alpha for `latest_release=1.0.0` and below it will create `new_alpha=1.0.1-alpha`.

Reply:
Ah gotcha, I missed that part. Thanks for clarifying.